src/Form/MessageSettingsForm.php

```diff
@@ -132,15 +128,15 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $config = $this->config('message.settings');
+    parent::submitForm($form, $form_state);
+
+    $config = $this->configFactory->getEditable('message.settings');
```

Comment:
@damiankloip got two questions you might know:

1. Why do we need the `getEditable()` here, where a [similar case](https://github.com/amitaibu/og/blob/8.x-1.x/og_ui/src/Form/AdminSettingsForm.php#L152) in OG doesn't?
1. How to deal with plugins that come with their own config? Do we want/ need to allow implementing modules to inject their values in Message's own [purge config](https://github.com/Gizra/message/blob/8.x-1.x/config/install/message.settings.yml#L2)? (which should probably move all under the `purge` key).